app/controllers/clearance/passwords_controller.rb

```diff
@@ -24,10 +27,12 @@ def new
   end
 
   def update
-    @user = find_user_for_update
+    @password_reset = find_password_reset_by_user_id_and_token
+    user = @password_reset.user
 
-    if @user.update_password password_reset_params
-      sign_in @user
+    if user.update_password password_reset_params
+      Clearance::PasswordResetDeactivator.new(user).run
```

Comment:
Hmm...we wouldn't want the resets to successfully deactivate if the password update failed, right? We could check for successful runs for both via:

```ruby
if user.update_password "whatever" && deactivator.run
``` 

...or something like that.

Was there some other flow you were thinking of?